src/encoder.rs

```diff
@@ -344,6 +346,53 @@ impl<T: Pixel> FrameState<T> {
     }
   }
 
+  pub fn compute_activity_mask(&mut self) {
+    //activity mask is computed only for luma plane
+
+    let luma_plane = &self.input.planes[0];
+    let PlaneConfig { width, height, .. } = luma_plane.cfg;
+    let luma = &luma_plane.data;
+    let PlaneConfig { xdec, ydec, .. } = self.activity_mask.cfg;
+    let mut act_mask = vec![0 as u32; (width >> xdec) * (height >> ydec) ];
+
+    for r in 0..(height >> ydec) {
+      for c in 0..(width >> xdec) {
+        let base = (r << xdec) * width + (c << ydec);
```

Comment:
The factor `width` here in calculating the `base` should be the stride of the luma plane and padding should also be considered. `Plane::index` encapsulates both of these concerns.
```suggestion
        let base = luma_plane.index(r << xdec, c << ydec);
```

Reply:
Wow, that's really clean! I'll make the changes accordingly.